subsys/shell/modules/date_service.c

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2020 Nick Ward
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+#include <shell/shell.h>
+#include <init.h>
+#include <string.h>
+#include <posix/time.h>
+#include <sys/timeutil.h>
+
+#define HELP_NONE "[none]"
+#define HELP_DATE_SET \
+	"<year: 1900-9999> <month: 1-12> <day: 1-31> " \
+	"<hour: 0-23> <minute: 0-59> <second: 0-60>"
+
+static void date_print(const struct shell *shell, struct tm *tm)
+{
+	shell_print(shell,
+		    "%d-%02d-%02d "
+		    "%02d:%02d:%02d UTC",
```

Comment:
Except for year these should use the `u` conversion specifier rather than `d` because they can't legitimately be negative.

Reply:
Done